src/App.js

```diff
@@ -0,0 +1,64 @@
+import React, {useState, useEffect} from 'react';
+import './style.css'
+import axios from 'axios'
+
+import {
+  BrowserRouter as Router,
+  Route,
+  Link
+}from 'react-router-dom'
+
+import StarshipPage from './StarshipPage'
+
+function App() {
+  const [starships, SetStarships] = useState('')
```

Comment:
The issue you're having might be from useState('').  If the initial state is set as an empty string rather than an empty array, then React will try to run the map on the empty string before the axios call updates the value of the starships array.  Try doing: useState([])